src/js/js.c

```diff
@@ -47,20 +47,48 @@ int js_get_code(jsval v) {
 	jsval code;
 
 	if (JS_GetProperty(js_context, JSVAL_TO_OBJECT(v), "code", &code)) {
-		printf("cod=%d\n", JSVAL_TO_INT(code));
 		return JSVAL_TO_INT(code);
 	}
 
 	return -1;
 }
 
-char* js_get_message(jsval v) {
-	jsval message;
+char *js_get_message(jsval v) {
+	jsval messages, msg, rval;
+	ssize_t messages_len;
 	char *c_str;
+	int i;
 
-	if (JS_GetProperty(js_context, JSVAL_TO_OBJECT(v), "message", &message)) {
-		c_str = JS_EncodeString(js_context, JSVAL_TO_STRING(message));
-		return c_str;
+	if (!JS_GetProperty(js_context, JSVAL_TO_OBJECT(v), "messages", &messages)) {
+		return NULL;
+	}
+
+	switch(JS_TypeOfValue(js_context, messages)) {
+		case JSTYPE_STRING:
+			c_str = JS_EncodeString(js_context, JSVAL_TO_STRING(msg));
+			return c_str;
+		case JSTYPE_OBJECT:
```

Comment:
Maybe it's better/easier to concatenate messages in C using the 'String Buffer' (string_tools.h). This would avoid repeated C/JS conversions.